R/duckdb_helpers.R

```diff
@@ -48,84 +49,76 @@
 #'   \item{year}{\code{double}. The year of the trip.}
 #' }
 #' @keywords internal
-spod_duckdb_od_v1 <- function(
+spod_duckdb_od <- function(
     con = DBI::dbConnect(duckdb::duckdb(), dbdir = ":memory:", read_only = FALSE),
     zones = c(
       "districts", "dist", "distr", "distritos",
       "municipalities", "muni", "municip", "municipios",
       "lau", "large_urban_areas", "gau", "grandes_areas_urbanas"
     ),
-    data_dir = spod_get_data_dir()) {
-  ver <- 1
-
+    ver = NULL,
+    data_dir = spod_get_data_dir()
+) {
+    
+  ver <- as.integer(ver)
+  if (!ver %in% c(1, 2)) {
+    stop("Invalid version number. Must be 1 or 2.")
+  }
+  
   zones <- match.arg(zones)
   zones <- spod_zone_names_en2es(zones)
 
-  csv_folder <- paste0(
-    data_dir, "/",
-    spod_subfolder_raw_data_cache(ver = ver),
-    "/maestra1-mitma-", spod_zone_names_en2es(zones),
-    "/ficheros-diarios/"
-  )
-
+  # check that gau is not selected with ver = 1
+  if (ver == 1 && zones == "gau") {
+    stop("The 'gau' zones are not available in v1 data.")
+  }
 
-  # create view of csv files and preset variable types
-  if (zones == "distritos") {
-    DBI::dbSendStatement(
-      con,
-      dplyr::sql(
-        glue::glue(
-          "CREATE VIEW od_csv_raw AS SELECT *
```

Comment:
Sooo much better with this as .sql files, right?

Reply:
Absolutely! And as I said, I love it how it made the internal helper function to create duckdb table "views" to csv folders much more concise. That was a great suggestion, @Robinlovelace !